carapace-server/src/main/java/org/carapaceproxy/server/config/SSLCertificateConfiguration.java

```diff
@@ -93,10 +93,13 @@ public boolean isMoreSpecific(SSLCertificateConfiguration other) {
         if (subjectAltNames == null || subjectAltNames.isEmpty()) {
             return hostname.length() > other.getHostname().length();
         }
-        final var otherNames = other.getNames().stream().map(CertificatesUtils::removeWildcard);
-        for (var n: getNames()) {
+
+        final var otherNamesList = other.getNames().stream().map(CertificatesUtils::removeWildcard)
+                .collect(Collectors.toList());
+
+        for (var n : getNames()) {
             final var name = CertificatesUtils.removeWildcard(n);
-            if (otherNames.anyMatch(on -> name.length() > on.length())) {
+            if (otherNamesList.stream().anyMatch(on -> name.length() > on.length())) {
```

Comment:
Here couldn't we just store the max length of the name in otherNamesList and check against it, instead cycle through `otherNamesList` each time?

Reply:
yes, make sense @NiccoMlt